src/librbd/internal.cc

```diff
@@ -3513,4 +3513,143 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
       ictx->perfcounter->inc(l_librbd_readahead_bytes, readahead_length);
     }
   }
+
+  // Consistency groups functions
+
+  int create_cg(librados::IoCtx& io_ctx, const char *cg_name)
+  {
+    ostringstream bid_ss;
+    uint32_t extra;
+    int remove_r;
+    string id, id_cg, header_oid;
+
+    CephContext *cct = (CephContext *)io_ctx.cct();
+
+    Rados rados(io_ctx);
+    uint64_t bid = rados.get_instance_id();
+
+    id_cg = util::id_cg_name(cg_name);
```

Comment:
I meant cg_header.98374. This is the object where we keep all the information about the consistency group. Maybe it makes sense to call it cg_header.cgname? Because then we can reach out to a consistency group without getting the directory.